examples/nlp/machine_translation/create_tarred_monolingual_dataset.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import os
+import pickle
+import tarfile
+import tempfile
+
+from nemo.collections.nlp.data import TranslationOneSideDataset
+from nemo.collections.nlp.modules.common.tokenizer_utils import get_tokenizer
+
+
+def write_batches_to_tarfiles(
+    args, fname, num_tokens, tokenizer, num_files_in_tar, tar_file_ptr, tar_file_ctr, global_batch_ctr,
+):
+    """
+    Writes current fragment of the overall parallel corpus to tarfiles by:
+    (1) Creating a minibatches using a TranslationDataset object.
+    (2) Writing each minibatch to a pickle file.
+    (3) Adding pickle files to a tarfile until it reaches args.num_batches_per_tarfile.
+    """
+
+    dataset = TranslationOneSideDataset(
+        tokenizer=tokenizer,
+        dataset=fname,
+        tokens_in_batch=num_tokens,
+        clean=args.clean,
+        max_seq_length=args.max_seq_length,
+        min_seq_length=args.min_seq_length,
+        cache_ids=False,
+    )
+
+    for batch in dataset.batches:
+        global_batch_ctr += 1
+        batch = {'src': batch}
+        pickle.dump(batch, open(os.path.join(args.out_dir, 'batch-%d.pkl' % (global_batch_ctr)), 'wb'))
+
+        if num_files_in_tar == args.num_batches_per_tarfile:
+            tar_file_ctr += 1
+            tar_file_ptr.close()
+            tar_file_ptr = tarfile.open(
+                os.path.join(args.out_dir, 'batches.tokens.%d.%d.tar' % (num_tokens, tar_file_ctr)), 'w'
+            )
+            num_files_in_tar = 0
+
+        tar_file_ptr.add(os.path.join(args.out_dir, 'batch-%d.pkl' % (global_batch_ctr)))
+        num_files_in_tar += 1
+        os.remove(os.path.join(args.out_dir, 'batch-%d.pkl' % (global_batch_ctr)))
+    return tar_file_ptr, global_batch_ctr, num_files_in_tar, tar_file_ctr
+
+
+if __name__ == '__main__':
```

Comment:
Can you please refactor this and move this logic to ``nemo.collections.nlp.data.machine_translation.preproc_mt_data.MTDataPreproc`` and just call a method of nemo.collections.nlp.data.machine_translation.preproc_mt_data.MTDataPreproc in this script?